core/src/main/java/com/amplifyframework/api/graphql/error/GraphQLPathSegment.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.api.graphql.error;
+
+import androidx.annotation.NonNull;
+
+public final class GraphQLPathSegment {
```

Comment:
Ah crumbs. When I did the checkstyle rules last year,  I think I had intended to include
[`MissingJavadocType`](https://checkstyle.sourceforge.io/config_javadoc.html#MissingJavadocType), so that we remembered to make Javadocs on public entities. I suspect we are missing some around the code base. Anyway, if a user will get this `GraphQLPathSegment` back in response, it would be nice to have doco for any `public` entity they might access. (Not expected to access constructors, but class-level doco may benefit the user?)

Reply:
Got it, I added Javadoc for this class and a few others as well.  If I get a chance later, I will follow up in a separate PR to add MissingJavadocType to checkstyle-rules.xml.